src/main/java/edu/wpi/grip/ui/PaletteView.java

```diff
@@ -49,6 +62,12 @@ public void clearOperations() {
 
     @Subscribe
     public void onOperationAdded(OperationAddedEvent event) {
-        this.operations.getChildren().add(new OperationView(this.eventBus, event.getOperation()));
+        OperationView view = new OperationView(this.eventBus, event.getOperation());
+        view.determineVisibility(this.operationSearch.getText());
```

Comment:
This is a little hacky.  Why not just have a `filterOperations` invalidation listener, and add it to both `this.getChildren()` and `this.operationSearch.textProperty()`?